piker/data/_web_bs.py

```diff
@@ -219,28 +219,33 @@ async def recv_task():
             '''
             receives every ws message and stores it in its corresponding result
             field, then sets the event to wakeup original sender tasks.
+            also, recieves responses to requests originated from the server side.
             '''
             async for msg in ws:
-                try:
-                    msg = response_type(**msg)
-
-                    if msg.id not in rpc_results:
-                        log.warning(f'Wasn\'t expecting ws msg: {json.dumps(msg, indent=4)}')
-
-                    res = rpc_results.setdefault(
-                        msg.id,
-                        {'result': None, 'event': trio.Event()}
-                    )
-
-                    res['result'] = msg
-                    res['event'].set()
-
-                except TypeError:
-                    if request_type == None:
-                        raise
-                    await request_hook(request_type(**msg))
-
-                
+                match msg:
+                    case {
+                        'result': _
+                    }:
+                        msg = response_type(**msg)
+
+                        if msg.id not in rpc_results:
+                            log.warning(f'Wasn\'t expecting ws msg: {json.dumps(msg, indent=4)}')
+
+                        res = rpc_results.setdefault(
+                            msg.id,
+                            {'result': None, 'event': trio.Event()}
```

Comment:
maybe simplify this to:

```python
res = rpc_results.setdefault(
    msg.id,
    {'result': msg, 'event': trio.Event()}
}
```

Also, i'm not entirely sure i follow the event setting here?
There either is, or isn't a waiting request right? In which case shouldn't you just assume the entry exists otherwise it's an error?

Reply:
Actually, just because I've gone through all the *tricks* with `match:` to make things more flat/succint, what y'all think about something more like this?

```python
match msg:
    case {
        'result': result,
        'id': mid,
    } if res_entry := rpc_results.get(mid):

        res_entry['result'] = response_type(**msg)
        res_entry['event'].set()

    case {
        'result': _,
        'id': mid,
    } if not rpc_results.get(mid):
        log.warning(f'Wasn\'t expecting ws msg: {json.dumps(msg, indent=4)}')

    case {
        'method': _,
        'params': _,
    } if request_hook:
        await request_hook(request_type(**msg))

    case _:
        log.warning(f'Unhandled JSON-RPC msg!?\n{msg}')
```

Reply:
😿 dat coo peeps?

Reply:
Yep, done!